### PR TITLE
Tests for OCastDomainGeneral

### PR DESCRIPTION
**Pull request by yanngodeau-orange**

### CODE REVIEW

File: OCastTests/TestOCastDomainGeneral.swift [hunk 1]
```diff
@@ -0,0 +1,109 @@
+//
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: add license

File: OCastTests/TestOCastDomainGeneral.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                    "trackId": "trackId",
+                    "enable": true
+                }
+            }
+        },
+        "type": "type",
+        "src": "source",
+        "dst": "destination"
+    }
+""".utf8)
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: indent problem ?

File: OCastTests/TestOCastDomainGeneral.swift [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    
+    override func setUp() {
+    }
+    
+    override func tearDown() {
+    }
+    
+    // Test that a missing key from the Json throwing an exception
+    private func assertThrowsKeyNotFound<T: Codable>(_ expectedKey: String, decoding: T.Type, from data: Data, file: StaticString = #file, line: UInt = #line) {
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: Maybe Decodable is enough

File: OCastTests/TestOCastDomainGeneral.swift [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        assertThrowsKeyNotFound("dst", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "dst"))
+    }
+    
+    func testDecodingWhenMissingSource() throws {
+        assertThrowsKeyNotFound("src", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "src"))
+    }
+    
+    
+    
+//    func testEncoding() {
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: remove if unused

File: OCastTests/TestOCastDomainGeneral.swift [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+//        let oCastApplicationLayer = OCastApplicationLayer(service: "service", data: oCastDataLayer)
+//        let oCastDeviceLayer = OCastDeviceLayer(source: "source", destination: "destination", id: 1, status: "status", type: "type", message: oCastApplicationLayer)
+//
+//        let jsonDataTest = try! JSONEncoder().encode(oCastDataLayer)
+//        let jsonStringTest = String(data: jsonDataTest, encoding: .utf8)!
+//        print("\n" + jsonStringTest + "\n")
+//    }
+    
+    
+    func testPerformanceExample() {
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: what is the interest of this test ?

File: OCastTests/TestOCastDomainGeneral.swift [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+        // This is an example of a performance test case.
+        self.measure {
+            // Put the code you want to measure the time of here.
+        }
+    }
+}
+
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: Key are tested, but never the values. Add "values" tests.

File: OCastTests/TestOCastDomainGeneral.swift [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+    
+    func testDecodingWhenMissingDestination() throws {
+        assertThrowsKeyNotFound("dst", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "dst"))
+    }
+    
+    func testDecodingWhenMissingSource() throws {
+        assertThrowsKeyNotFound("src", decoding: OCastDeviceLayer<MediaTrackCommand>.self, from: try jsonData.json(deletingKeyPaths: "src"))
+    }
+    
+    func testDecodingWhenWringValue() throws {
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: what is a "Wring" value ?
this test seem useless.

File: OCastTests/TestOCastDomainGeneral.swift [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+    
+    func testEncoding() throws {
+        let jsonDataTest = try! JSONEncoder().encode(jsonObject)
+        let jsonStringTest = String(data: jsonDataTest, encoding: .utf8)!
+        let trimmedTest = jsonStringTest.replacingOccurrences(of: "\\s*", with: "", options: .regularExpression)
+        
+        XCTAssertNotNil(trimmedTest.range(of: jsonRegexPattern, options: .regularExpression, range: nil, locale: nil))
+    }
+    
+    func testDecoding() {
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: each function must be tested for proper operation but also its case errors. You need to test a fail decoding.

File: OCastTests/TestOCastDomainGeneral.swift [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+            if case .keyNotFound(let key, _)? = error as? DecodingError {
+                XCTAssertEqual(expectedKey, key.stringValue, "Expected missing key '\(key.stringValue)' to equal '\(expectedKey)'.", file: file, line: line)
+            } else {
+                XCTFail("Expected '.keyNotFound(\(expectedKey))' but got \(error)", file: file, line: line)
+            }
+        }
+    }
+}
+
+extension OCastDeviceLayer {
```
Comment (chazemar) [inline OCastTests/TestOCastDomainGeneral.swift]: If we extend a class that is tested,  does this not alter the behavior of the tested class?

Comment (yanngodeau-orange) [reply OCastTests/TestOCastDomainGeneral.swift]: The purpose of these classes is to test the equality between two DeviceLayer objects. They do not alter the behavior of the tested class and are only used in the context of this test.

**Comment by chazemar** [comment]
LGTM

**Pull request closed without merging**